integration-tests/deployment/address_book.go

```
@@ -109,9 +110,6 @@ func (m *AddressBookMap) Save(chainSelector uint64, address string, typeAndVersi
 		// First time chain add, create map
 		m.AddressesByChain[chainSelector] = make(map[string]TypeAndVersion)
 	}
-	if _, exists := m.AddressesByChain[chainSelector][address]; exists {
```

@AnieeG @connorwstein  i needed to change this. i think it's safe since it's normal for merge to overwrite and existing addr. problem was uncovered if i wanted to re-run a migration in `chainlink-deployments`

---

The intention with the strictness here was to avoid silently failing in this scenario: someone writes a buggy changeset which was supposed to deploy a new address, but it returns an existing address instead. In that case the changeset will succeed but the user is unaware of the issue. To iterate on changesets you can:
1. Run changeset A, generates artifacts/A...
2. Inspect artifacts, discover an issue. Remove artifacts/A...
3. Repeat 1-2 until satisfied. 
4. Promote artifacts via CLI/CI.